groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/completion/inference/TypeInferenceVisitor.java

```diff
@@ -194,4 +193,29 @@ private static boolean sameVariableName(ASTNode node1, FieldNode node2) {
                 && ((VariableExpression) node1).getName().equals(node2.getName());
     }
 
+    private ClassNode deriveExpressonType(Expression expression) {
```

Comment:
What abbout a single return statement here? IMHO those are too many. It's also a bit inconsistent with the previous method (visitDeclarationExpression).